type-example.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Foo;
+
+use ProxyManager\Factory\AccessInterceptorScopeLocalizerFactory;
+use ProxyManager\Factory\AccessInterceptorValueHolderFactory;
+use ProxyManager\Factory\LazyLoadingGhostFactory;
+use ProxyManager\Factory\LazyLoadingValueHolderFactory;
+use ProxyManager\Factory\NullObjectFactory;
+use ProxyManager\Factory\RemoteObject\AdapterInterface;
+use ProxyManager\Factory\RemoteObjectFactory;
+use ProxyManager\Proxy\GhostObjectInterface;
+use ProxyManager\Proxy\LazyLoadingInterface;
+
+require_once __DIR__ . '/vendor/autoload.php';
+
+class MyProxiedClass
+{
+    public function sayHello() : string
+    {
+        return 'Hello!';
+    }
+}
+
+echo (new AccessInterceptorScopeLocalizerFactory())
+    ->createProxy(
+        new MyProxiedClass(),
+        ['sayHello' => static function (string $foo) { 'ha'; }]
+    )
+    ->sayHello();
+
+echo (new AccessInterceptorValueHolderFactory())
+    ->createProxy(new MyProxiedClass())
+    ->sayHello();
+
+echo (new LazyLoadingGhostFactory())
+    ->createProxy(MyProxiedClass::class,
+        $initializer = function (
+            GhostObjectInterface $proxy,
+            string $method,
+            array $parameters,
+            & $initializer,
+            array $properties
+        ) {
+            $initializer = null; // disable initialization
+        })
+    ->sayHello();
```

Comment:
This type declaration seems to fail type checks:

```
ERROR: UndefinedInterfaceMethod - type-example.php:49:7 - Method ProxyManager\Proxy\GhostObjectInterface::sayhello does not exist
    ->sayHello();
```

@muglug I'm sorry to chase you across repositories, but do you maybe know whether this is a docblock parser bug? This is the docblock:

https://github.com/Ocramius/ProxyManager/blob/300973221ee90f0f126e5f3b2753410c7c5c88d4/src/ProxyManager/Factory/LazyLoadingGhostFactory.php#L38-L88

Reply:
that docblock should be
```php
/**
 * @psalm-template RealObjectType 
 * 
 * @psalm-param class-string<RealObjectType> $className 
 * @psalm-param array{skippedProperties?: array<int, string>} $proxyOptions
 * 
 * @psalm-return RealObjectType&GhostObjectInterface 
 */ 
```

Reply:
Daaang, so it is just that `?`!

Reply:
And missing `$proxyOptions`